tests/qc_checkers/test_cluster_pf.py

```diff
@@ -0,0 +1,107 @@
+from collections import namedtuple
+import pytest
+
+from checkQC.qc_checkers.cluster_pf import cluster_pf
+from checkQC.handlers.qc_handler import QCErrorFatal, QCErrorWarning
+
+@pytest.fixture
+def qc_data():
+    return namedtuple("QCData", "sequencing_metrics")(
+        {
+            1: {"total_cluster_pf": 1000},
+            2: {"total_cluster_pf": 10},
+            3: {"total_cluster_pf": 100},
+            4: {"total_cluster_pf": 10000},
+        }
+    )
+
+
+EXP_MSG = "Clusters PF {total_cluster_pf} > {threshold} on lane {lane}"
+
+
+def test_cluster_pf(qc_data):
+    qc_reports = cluster_pf(
+        qc_data,
+        error_threshold=50,
+        warning_threshold=500,
+    )
+
+    assert len(qc_reports) == 2
+    for report in qc_reports:
+        lane, threshold = report.data["lane"], report.data["threshold"]
+        match lane:
+            case 2:
+                exp_data = {
+                    "total_cluster_pf": qc_data.sequencing_metrics[lane]["total_cluster_pf"],
+                    "threshold": 50,
+                    "lane": lane,
+                }
+                assert report.message == EXP_MSG.format(**exp_data)
+                assert report.type() == "error"
+                assert report.data == exp_data
+            case 3:
+                exp_data = {
+                    "total_cluster_pf": qc_data.sequencing_metrics[lane]["total_cluster_pf"],
+                    "threshold": 500,
+                    "lane": lane,
+                }
+                assert report.message == EXP_MSG.format(**exp_data)
+                assert report.type() == "warning"
+                assert report.data == exp_data
+            case _:
+                raise ValueError(f"Lane {lane} didn't pass QC check.")
```

Comment:
Just a suggestion, but I think it could be clearer for the developer if the message was: "Lane unexpectedly did not pass QC check" or similar, since it is expected in some cases. Might be overkill though.